classes/SupportManager.php

```diff
@@ -385,163 +262,158 @@ public function supportSearch($variables = [], $limit = false, $order = [], $off
+                ];
+                break;
+            default:
+                $criteria = [
+                    'open' => 1,
+                    'admin_response' => 0,
+                    'support_type' => 'IN_' . $this->processTypes(true)
+                ];
         }
-        else {
-            return false;
+
+        return $this->supportSearch($criteria, $limit, ['updated' => $order], $offset);
+    }
+
+    public function fetchUserSupports() {
+        if(!$this->user) {
+            return [];
         }
+
+        $supports = $this->supportSearch(['user_id'=>$this->user_id], 50, ['open'=>'DESC', 'updated'=>'DESC']);
+
+        return ($supports ?? []);
     }
 
-    /**
-     * @param $support_id
-     * @return bool
-     * Allows a user to assign their guest support to their account, should they want to follow up further
-     * through their account.
-     */
-    public function assignGuestSupportToUser($support_id) {
-        $data = $this->supportSearch(['support_id'=>$support_id], 1, ['time'=>'DESC'])[0];
+    public function fetchSupportByID($support_id, $forceGuest = false) {
+        $support_id = (int) $support_id;
 
-        if(!$data) {
-            return false;
-        }
-        if($data['user_id'] != 0) {
-            //return false;
+        $criteria = ['support_id' => $support_id];
+
+        //If not staff, restrict to user id
+        if(!$this->staff && !$forceGuest) {
+            $criteria['user_id'] = $this->user_id;
         }
 
-        if($this->system->query("UPDATE `support_request` SET `user_id`='{$this->user_id}' WHERE `support_id`='{$support_id}' LIMIT 1")) {
-            $this->addSupportResponses($support_id, $this->user_id, $data['user_name'], "Added to my account.", $_SERVER['REMOTE_ADDR']);
-            return true;
+        $result = $this->supportSearch($criteria, 1);
+        return($result ? $result[0] : false);
+    }
+
+    public function fetchSupportResponses($support_id) {
+        $result = $this->system->query("SELECT * FROM `support_request_responses` 
+            WHERE `support_id`='{$support_id}' ORDER BY `time` DESC, `response_id` DESC");
+        if($this->system->db_last_num_rows) {
+            $responses = [];
+            while($row = $this->system->db_fetch($result)) {
+                $responses[] = $row;
+            }
+            return $responses;
         }
         return false;
     }
 
-    /**
-     *
-     *
-     *
-     * @param  int        $support_id
-     * @return bool
-     * @throws Exception
-     */
-    public function updateSupport($support_id, $admin_id = 0, $admin_name = '', $staff_level=0) {
-        // Validate support exists and belongs to user
-        if($this->admin) {
-            $support_data = $this->fetchSupportByID($support_id);
-        }
-        else {
-            $support_data = $this->fetchSupportByID($support_id, $this->user_id);
-        }
+    public function updateSupport($support_id, $support_staff = false) {
```

Comment:
Could we replace the `$support_staff` parameter with `$this->user->isSupportStaff()`?

Reply:
Will change this to `$this->staff` as that dictates if the instance of SupportManager is a staff connection or a user connection. 

My thought process here is:
Oreo has a support request entered and is support staff. Replies on the user end, I'd think sending a PM that her support was updated could be confusing.

Reply:
That makes sense - We can go ahead and merge this, but we should update that variable to reflect that. Possible names:

`$this->in_staff_mode`
`$this->viewing_as_staff`